examples/asyncapi-from-parser/index.ts

```diff
@@ -0,0 +1,39 @@
+import { parse } from '@asyncapi/parser';
+import { TypeScriptGenerator } from '../../src';
+
+const generator = new TypeScriptGenerator({ modelType: 'interface' });
+const AsyncAPIDocument = {
+  asyncapi: '2.1.0',
+  info: {
+    title: 'example',
+    version: '0.1.0'
+  },
+  channels: {
+    '/test': {
+      subscribe: {
+        message: {
+          payload: {
+            $schema: 'http://json-schema.org/draft-07/schema#',
+            type: 'object',
+            additionalProperties: false,
+            properties: {
+              email: {
+                type: 'string',
+                format: 'email'
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+};
+
+export async function generate(logCallback : (msg: string) => void) : Promise<void> {
```

Comment:
sorry to be a pain here but I strongly thing this console.log callback is polluting the example. We should do this:

```js
export async function generate() : Promise<void> {
  const parsedDoc = await parse(JSON.stringify(AsyncAPIDocument));
  const models = await generator.generate(parsedDoc as any);
  for (const model of models) {
    console.log(model.result);
  }
}

generate();
```

and test should do
```js
  test('and should log expected output to console', async () => {
    console.log = jest.fn();
    await generate();
    expect(console.log).toMatchSnapshot();
  });
```

I think it is much cleaner

Reply:
I think there was a reason why I chose to use callback, but cannot remember it anymore 😆 Changed!